lemur/utility.py

```diff
@@ -837,49 +828,185 @@ def delete_class(class_id):
     ds.commit()
 
 
-# change the students in a class with class id
-def change_class_students(class_id, student_names):
-    # clean up all the white space in the string
-    student_names_str = ''.join(student_names.split())
-    # get role objects from the Role table
-    role_student = get_role('Student')
-    # Process the string to strip all the white space;
-    # then split it over commas and join with commas
-    student_names = student_names_str.split(',')
-    # Check the correctness of the string to be strings delimited only by comma
-    pattern_for_student_names = '[a-zA-Z](,[a-zA-Z])*'
-    if (re.match(pattern_for_student_names, student_names_str) is None):
-        err_msg = 'The pattern for student names is not correct'
-        return err_msg
-    student_ids = [generate_user_id(s) for s in student_names]
+# Change the users(both professors and students) in a class
+def change_class_users(class_id, new_users):
+    if not class_exists(class_id):
+        return 'Class with id: {} doesn\'t exist'.format(class_id)
     the_class = get_class(class_id)
+    old_users = the_class.users
+    the_class.users = []
+    # Add new users to the class;
+    # add the associated labs to these users lab list
+    for u in new_users:
+        if not user_exists(str(u)):
+            ds.rollback()
+            return 'User with username: {} doesn\'t exist'.format(u)
+        else:
+            user = get_user(u)
+            user.labs = the_class.labs
+            the_class.users.append(user)
+    ds.commit()
```

Comment:
is this happening in two different transactions now? It seems weird to call ds.commit twice for this one operation.